File: c#/Fibonacci.cs
using System;
using System.Diagnostics;
using System.Linq;

namespace CSharp
{
    public class Fibonacci
    {
        public static long f(long n)
        {
            if (n <= 2)
            {
                return 1;
            }
            long n1 = 1;
            long n2 = 1;
            long result = 0;

            for (long i = 3; i <= n; i++)
            {
                result = n1 + n2;
                n2 = n1;
                n1 = result;
            }
            return result;
        }

        public static void trackExecutionSpeed()
        {
            Console.WriteLine("Average time to execute f(90) in nanoseconds");
            const int EXECUTION_TIMES = 200;
            const int CAL_FAB_TIMES = 50_000;
            double[] results = new double[EXECUTION_TIMES];
            Stopwatch sw = new Stopwatch();
            for (int i = 0; i < EXECUTION_TIMES; i++)
            {
                //long startTime = System.nanoTime();
                sw.Restart();
                for (int j = 0; j < CAL_FAB_TIMES; j++)
                {
                    f(90);
                }
                sw.Stop();
                //long totalTime = sw.ElapsedMilliseconds;
                var totalTimeSpan = sw.Elapsed;
                double totalTime = totalTimeSpan.TotalMilliseconds * 1000000;
                results[i] = totalTime/CAL_FAB_TIMES;
            }

            //Console.WriteLine($"C# Result: Avarage={results.Average()}ms, Detail={String.Join(",", results)}");

            Console.WriteLine($"C# Result: Avarage={results.Average()}ms");
        }

        public static void Main(string[] args)
        {
            trackExecutionSpeed();
        }
    }
}

File: java/Fibonacci.java
import java.util.Arrays;
import java.util.OptionalDouble;
import java.util.stream.Collectors;

/**
 * Class used to calculate the fibonacci of some number many times and each time
 * print the total time it took to calculate the value
 *
 * @author Rodrigo Ramirez
 */
public class Fibonacci {

    /**
     * Calculates the Fibonacci of n :
     * @param n
     * @return f(n)
     */
    public static long f(long n) {
        if (n <= 2) {
            return 1;
        }
        long n1 = 1;
        long n2 = 1;
        long result = 0;

        for(long i = 3; i <= n; i++ ) {
            result = n1 + n2;
            n2 = n1;
            n1 = result;
        }
        return result;
    }


    /**
     * Prints the average time it takes to calculate f(n)
     * @param args
     */
    public static void trackExecutionSpeed() {
        System.out.println("Average time to execute f(90) in nanoseconds");
        final int EXECUTION_TIMES = 200;
        final int CAL_FAB_TIMES = 50000;
        double[] results = new double[EXECUTION_TIMES];
        for(int i = 0; i < EXECUTION_TIMES; i++) {
            long startTime = System.nanoTime();
            for(int j = 0; j < CAL_FAB_TIMES; j++) {
                f(90);
            }
            long totalTime = System.nanoTime() - startTime;
            results[i] = totalTime/CAL_FAB_TIMES;
        }

        // System.out.println(String.format(
        //     "Java Result: Avarage=%fms, Detail=%s",
        //     Arrays.stream(results).average().orElse(0.0D),
        //     String.join(",", Arrays.stream(results).boxed().map(d -> d.toString()).toArray(String[]::new)))
        // );

        System.out.println(String.format(
            "Java Result: Avarage=%fms",
            Arrays.stream(results).average().orElse(0.0D))
        );
    }

    public static void main(String[] args) {
        trackExecutionSpeed();
    }
}
